presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGenerator.java

```diff
@@ -392,6 +393,47 @@ else if (fractionInput instanceof VariableReferenceExpression) {
             return format("PERCENTILEEST%d(%s)", percentile, inputSelections.get(getVariableReference(inputs.get(0))));
         }
 
+        private String handleApproxDistinct(CallExpression aggregation, Map<VariableReferenceExpression, Selection> inputSelections)
+        {
+            List<RowExpression> inputs = aggregation.getArguments();
+            if (inputs.isEmpty() || inputs.size() > 2) {
+                throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), "Cannot handle approx_distinct function " + aggregation);
+            }
+            Selection selection = inputSelections.get(getVariableReference(inputs.get(0)));
+            if (inputs.size() == 1) {
+                return format("DISTINCTCOUNTHLL(%s)", selection);
+            }
+            RowExpression standardErrorInput = inputs.get(1);
+            String standardErrorString;
+            if (standardErrorInput instanceof ConstantExpression) {
+                standardErrorString = getLiteralAsString((ConstantExpression) standardErrorInput);
+            }
+            else if (standardErrorInput instanceof VariableReferenceExpression) {
+                Selection fraction = inputSelections.get(standardErrorInput);
+                if (fraction.getOrigin() != LITERAL) {
+                    throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(),
+                            "Cannot handle approx_distinct standard error argument be a non literal " + aggregation);
+                }
+                standardErrorString = fraction.getDefinition();
+            }
+            else {
+                throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), "Expected the standard error to be a constant or a variable " + standardErrorInput);
+            }
+
+            double standardError = Double.parseDouble(standardErrorString);
+            if (standardError < 0) {
+                throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(),
+                        format("Cannot handle approx_distinct parsed as %d from input %s (function %s)", standardError, standardErrorString, aggregation));
+            }
+            // Pinot uses DisctintCountHll to do distinct count estimation, with hyperloglog algorithm.
```

Comment:
typo ... DistinctCountHll, can you also capitalize that to be consistent. 

Can you add some more details on the second argument that is needed: What is log2m in english would help understand the formula on line 433

Reply:
done